profile-widget/profilewidget2.h

```diff
@@ -227,8 +228,17 @@ public
 
 	//specifics for ADD and PLAN
 #ifndef SUBSURFACE_MOBILE
-	// Use std::vector<> and std::unique_ptr<>, because QVector<QScopedPointer<...>> is unsupported.
-	std::vector<std::unique_ptr<DivePictureItem>> pictures;
+	// The list of pictures in this plot. The pictures are sorted by offset in seconds.
+	// For the same offset, sort by filename.
+	// Pictures that are outside of the dive time are not shown. In this case, thumbnail
+	// is a null-pointer.
+	struct PictureEntry {
+		offset_t offset;
+		QString filename;
+		std::unique_ptr<DivePictureItem> thumbnail;
+		bool operator< (const PictureEntry &e) const;
```

Comment:
what does it mean to have an operator as part of a struct?

Reply:
it allows two `PictureEntry` structs to be compared, in the lines of `pictureEntry1 < pictureEntry2`, based on the `bool ProfileWidget2::PictureEntry::operator< (const PictureEntry &e) const` operator implementation.

it isn't considered a bad practice to have operators and methods for structs in C++. the language allows that.

Reply:
It means that the object can be compared with `<`. So you can write
```
	if (e1 < e2) {
		...
	}
```

Here, this is used so that we can sort the images with a default function. As an alternative, we could supply either a lambda or an explicit function to the `sort()` call.